### PR TITLE
Destination Iceberg v2: basic test scaffolds + related runtime classes

### PR DESCRIPTION
**Pull request by edgao**
add the spec test. Running this test class autogenerates the `expected-spec-*.json` files using whatever is defined in IcebergV2Specification.

and also https://github.com/airbytehq/airbyte/pull/48546 + https://github.com/airbytehq/airbyte/pull/48547

### CODE REVIEW

File: airbyte-integrations/connectors/destination-iceberg-v2/src/test-integration-legacy/kotlin/io/airbyte/integrations/destination/iceberg_v2/IcebergV2SpecTest.kt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package io.airbyte.integrations.destination.iceberg_v2
+
+import io.airbyte.cdk.load.spec.SpecTest
+
+class IcebergV2SpecTest : SpecTest()
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-iceberg-v2/src/test-integration-legacy/kotlin/io/airbyte/integrations/destination/iceberg_v2/IcebergV2SpecTest.kt:5]: fyi: there's probably some way to get JUnit to run the base `SpecTest` directly, but I couldn't get it working, and writing this class is easy enough that I didn't particularly care :P

File: airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/load/spec/SpecTest.kt [hunk 2]
```diff
@@ -80,7 +84,7 @@ abstract class SpecTest :
         val jsonMatcher: JsonMatcher =
             CompositeJsonMatcher(
                 StrictJsonArrayPartialMatcher(),
-                LenientJsonObjectPartialMatcher(),
+                StrictJsonObjectPartialMatcher(),
```
Comment (edgao) [inline airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/load/spec/SpecTest.kt:87]: previously this wasn't detecting new properties, switching to the strict matcher gives us the behavior we want

File: airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/load/spec/SpecTest.kt [hunk 3]
```diff
@@ -80,15 +84,16 @@ abstract class SpecTest :
...
-        val diff: JsonDiff =
+        val diff = OnlyErrorDiffViewer.from(
             DiffGenerator.diff(expectedSpec, Jsons.writeValueAsString(spec), jsonMatcher)
+        ).toString()
         assertAll(
             "Spec snapshot test failed. Run this test locally and then `git diff <...>/$expectedSpecFilename` to see what changed, and commit the diff if that change was intentional.",
-            { Assertions.assertEquals("", OnlyErrorDiffViewer.from(diff).toString()) },
-            { Assertions.assertEquals(expectedSpec, actualSpecPrettyPrint) }
+            { Assertions.assertTrue(diff.isEmpty(), "Detected semantic diff in JSON:\n" + diff.prependIndent("\t\t")) },
+            { Assertions.assertTrue(expectedSpec == actualSpecPrettyPrint, "File contents did not equal generated spec, see git diff for details") }
```
Comment (edgao) [inline airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/load/spec/SpecTest.kt:96]: making the test output more readable

**Comment by jdpgrailsdev** [comment]
@edgao Looks like there is a formatting issue with SpecTest.kt

**Review by jdpgrailsdev — APPROVED** [review]
:shipit: once the formatting issue is resolved.

**Pull request merged**